### ISSUE TITLE
waydroid init download gets stuck at anytime

### ISSUE THREAD

**Issue by unknownpersonog** [waydroid/waydroid#651]
When I run``sudo waydroid init`` or run the waydroid app and start the download either GAPPS or VANILLA the download will work fine at start but randomly stops at anytime
By stop I mean no change in download speed and downloaded data It just hangs there!!
I tried it like 10 times still same. I did restart my pc

**Comment by cloud-owen** [comment]
i have the same problem

**Comment by iTzVoko** [comment]
Can I download it manually? (i.e. use my own download manager)

**Comment by charlesthobe** [comment]
This is reproducible even when downloading the image through a browser, I think a possible solution would be hosting the images somewhere else, or praying that sourceforge fixes whatever issue with their server.

![Screenshot from 2023-09-07 02-42-46](https://github.com/waydroid/waydroid/assets/95266557/f5c86b97-d604-4584-af81-c948af93c76f)

**Comment by charlesthobe** [comment]
some days it works, others, it doesn't.

**Comment by charlesthobe** [comment]
> Ditto. The bigger issue is that when I download it again, it starts from the zero! Can I download it manually? (i.e. use my own download manager)

yes go to https://sourceforge.net/projects/waydroid/files/images/
and download a system image and a vendor image of matching versions, and extract the files to `/var/lib/waydroid/images/` so it is:

`/var/lib/waydroid/images/system.img`
and
`/var/lib/waydroid/images/vendor.img`

@iTzVoko 
although be careful since it still fails with the browser at 99%, so you're gonna need to find a way of continuing the download, like using google chrome's "Copy as cURL" option and using `curl` with the `-C` flag to continue.

**Comment by iTzVoko** [comment]
@charlesthobe
>  and download a system image and a vendor image of matching versions, and extract the files to /var/lib/waydroid/images/ so it is:

>/var/lib/waydroid/images/system.img
> and
> /var/lib/waydroid/images/vendor.img

Thank you. I was missing these paths.

>  although be careful since it still fails with the browser at 99%, so you're gonna need to find a way of continuing the download, like using google chrome's "Copy as cURL" option and using curl with the -C flag to continue.
actually nvm, you can use wget with -c and just the download link, curl isn't needed.

Yep ;)
cURL is a more versatile tool than Wget.

**Comment by cyfex** [comment]
> @charlesthobe
> 
> > and download a system image and a vendor image of matching versions, and extract the files to /var/lib/waydroid/images/ so it is:
> 
> > /var/lib/waydroid/images/system.img
> > and
> > /var/lib/waydroid/images/vendor.img
> 
> Thank you. I was missing these paths.
> 
> > although be careful since it still fails with the browser at 99%, so you're gonna need to find a way of continuing the download, like using google chrome's "Copy as cURL" option and using curl with the -C flag to continue.
> > actually nvm, you can use wget with -c and just the download link, curl isn't needed.
> 
> Yep, I agree ;) cURL is a more versatile tool than Wget.

How do I init with my manually downloaded images? After the download and placing in /var/lib/waydroid/images, when I waydroid init, it still tries to download from the scratch.

**Comment by charlesthobe** [comment]
@cyfex 
I apologize I only referenced my already done installation so I didn't know you would still have to init.
Please follow these [instructions](https://docs.waydro.id/faq/using-custom-waydroid-images)
> First, download the images manually from sourceforge, or use your custom built system.img and vendor.img produced from following the [Compile Instructions](https://github.com/development/compile-waydroid-lineage-os-based-images#how-to-build). Then copy/move them to the following folder:
/usr/share/waydroid-extra/images/
 Then we need to re-init Waydroid in order for it to use the custom images:
waydroid init -f

I just tested and it works, you may have to create the directory first (I did):
```
mkdir -p /usr/share/waydroid-extra/images/
```

**Comment by Dvaranir** [comment]
Any1 who have problems with downloading from sourceforge https://motrix.app If I understand correct, this thing is downloading files with multiple threads. Download speed is fast and it can continue download if something happens

**Comment by adhamsalama** [comment]
Same here.
Download is stuck at `[Downloading]  767.3 MB/767.37 MB    1712.22 kbps(approx.) `

**Comment by mrxdoffical** [comment]
> Same here. Download is stuck at `[Downloading] 767.3 MB/767.37 MB 1712.22 kbps(approx.) `

got the same issue did you actually fix it

**Comment by adhamsalama** [comment]
No. I decided I don't care.

**Comment by ghost** [comment]
> Same here. Download is stuck at `[Downloading] 767.3 MB/767.37 MB 1712.22 kbps(approx.) `

شغل vpn وجرب تحمل تاني المفروض ده يحل المشكلة إن شاء الله

وحمله من الموقع الرسمي من هنا: https://docs.waydro.id/usage/install-on-desktops

بس شغل vpn

------------------

شغل أي vpn انت عايزه..

بس أنا استخدمت ده: https://youtube.com/shorts/HJLUYOaYTcE?si=OxHVdBNk1D6l4y_D

واستخدم الtcp 

مش ال udp

أنا استخدمت tcp443 

just use vpn and try to download again, this will fix the issue inshallah
